velox/functions/sparksql/String.h

```diff
@@ -960,12 +960,16 @@ struct ConvFunction {
   }
 
   // For signed value, toBase is negative.
-  static void
-  toChars(out_type<Varchar>& result, int64_t signedValue, int32_t toBase) {
+  static void toChars(
+      out_type<Varchar>& result,
+      int64_t signedValue,
+      int32_t toBase,
+      bool isMinInt64Num) {
+    int64_t absValue = UNLIKELY(isMinInt64Num)
```

Comment:
I find this logic confusing. Technically it is incorrect to say that abs(std::numeric_limits<int64_t>::min()) is std::numeric_limits<int64_t>::max(). I guess here it happens to work because the value is used only to compute resultSize.

It might be less confusing if we move resultSize computation outside of this function and change the signature to take result, signedValue, toBase and resultSize.

Notes section for reference: https://en.cppreference.com/w/cpp/numeric/math/abs

Reply:
thank you for the review, I addressed the comments